src/main/java/org/terasology/lost/OnSpawnSystem.java

```diff
@@ -15,26 +15,139 @@
+        progressTrackingComponent.addLevel("Temperate rain forest", "Lost:well");
+        progressTrackingComponent.addLevel("Temperate deciduous forest", "Lost:well");
+        progressTrackingComponent.addLevel("Subtropical desert", "Lost:well");
+        progressTrackingComponent.addLevel("Shrubland", "Lost:well");
+        progressTrackingComponent.addLevel("Marsh", "Lost:well");
+        progressTrackingComponent.addLevel("Tropical rain forest", "Lost:well");
+        progressTrackingComponent.addLevel("Tropical seasonal forest", "Lost:well");
+        progressTrackingComponent.addLevel("Lakeshore", "Lost:well");
+        progressTrackingComponent.addLevel("Coast", "Lost:well");
+        progressTrackingComponent.addLevel("Ice", "Lost:well");
+        player.addComponent(progressTrackingComponent);
+    }
+
+    @ReceiveEvent
+    public void onBiomeChange(OnBiomeChangedEvent event, EntityRef player) {
+        console.addMessage("OLD:" + event.getOldBiome().getDisplayName());
+        console.addMessage("NEW:" + event.getNewBiome().getDisplayName());
+        LocationComponent loc = player.getComponent(LocationComponent.class);
+        Vector3f pos = loc.getWorldPosition();
+        ProgressTrackingComponent progressTrackingComponent = player.getComponent(ProgressTrackingComponent.class);
+        int searchRadius = 0;
+        Vector3i ext = new Vector3i(searchRadius, 0, searchRadius);
+        Vector3i desiredPos = new Vector3i(pos.getX(), 1, pos.getZ());
+
+        // try and find somewhere in this region a spot to land
+        Region3i spawnArea = Region3i.createFromCenterExtents(desiredPos, ext);
+        Region worldRegion = LostWorldGenerator.world.getWorldData(spawnArea);
+
+        GraphFacet graphs = worldRegion.getFacet(GraphFacet.class);
+        WhittakerBiomeModelFacet model = worldRegion.getFacet(WhittakerBiomeModelFacet.class);
+        Vector2f pos2d = new Vector2f(pos.getX(), pos.getZ());
+        double d = 1000000;
+        ImmutableVector2f center = null;
+        String biomeName = null;
+        for (Graph g : graphs.getAllGraphs()) {
+            BiomeModel biomeModel = model.get(g);
+            for (org.terasology.polyworld.graph.Region r : g.getRegions()) {
+                WhittakerBiome biome = biomeModel.getBiome(r);
+                if (biome.getDisplayName().contains(event.getNewBiome().getDisplayName())) {
+                    double temp = Math.pow(pos2d.x - pos.getX(), 2) + Math.pow(pos2d.y - pos.getZ(), 2);
+                    if (temp < d) {
+                        d = temp;
+                        center = r.getCenter();
+                        biomeName = biome.getDisplayName();
+                    }
+                }
+
+            }
+        }
+        console.addMessage(center.toString());
+        if ((!progressTrackingComponent.isWellFound())&&progressTrackingComponent.getLevelPrefab(biomeName).contains("well")) {
```

Comment:
Trivial: code style issue to not have spaces around the `&&` - you might want to make sure you have Checkstyle installed in your IDE so it'll highlight these issues, easy to fix :-)